onnxruntime/core/providers/rocm/tunable/gemm_hipblaslt.h

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#ifdef USE_HIPBLASLT
+#include <hipblaslt/hipblaslt.h>
+#endif
+
+#include "core/common/common.h"
+#include "core/providers/rocm/tunable/gemm_common.h"
+#include "core/providers/rocm/tunable/rocm_tunable.h"
+
+namespace onnxruntime {
+namespace rocm {
+namespace tunable {
+namespace blas {
+namespace internal {
+
+#ifdef USE_HIPBLASLT
+
+template <typename T>
+constexpr hipblasDatatype_t HipBlasDataTypeFor(const T*) {
+  static_assert(sizeof(T) == 0, "Unsupported data type for hipBLASLt operation.");
+  // Compiler will complain if we don't return something.
+  return HIPBLAS_R_32F;
+}
+
+template <>
+constexpr hipblasDatatype_t HipBlasDataTypeFor(const float*) {
+  return HIPBLAS_R_32F;
+}
+
+template <>
+constexpr hipblasDatatype_t HipBlasDataTypeFor(const half*) {
+  return HIPBLAS_R_16F;
+}
+
+template <>
+constexpr hipblasDatatype_t HipBlasDataTypeFor(const BFloat16*) {
+  return HIPBLAS_R_16B;
+}
+
+template <>
+constexpr hipblasDatatype_t HipBlasDataTypeFor(const double*) {
+  return HIPBLAS_R_64F;
+}
+
+template <typename T, typename ParamsT>
+class HipBlasLtGemmOp {
+ public:
+  Status operator()(const ParamsT* params) {
+
+    hipblasLtHandle_t handle;
+    HIPBLASLT_CALL_THROW(hipblasLtCreate(&handle));
```

Comment:
Do not use `*_THROW` macros in function that returns `Status`, use the `HIPBLASLT_RETURN_IF_ERROR` variant.